static/js/apps/nl_interface/nl_commentary.tsx

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Commentary for NL results.
+ */
+
+import React from "react";
+
+import { SearchResult } from "../../types/app/nl_interface_types";
+
+interface NLCommentaryPropType {
+  chartsData: SearchResult;
+  hideCharts: boolean;
+  setHideCharts: (v: boolean) => void;
+}
+
+export function shouldHideCharts(respData: any): boolean {
+  const fb = respData["placeFallback"];
+  // If there's a valid fallback or if the place source is DEFAULT but not Earth,
+  // we should hide.
+  return (
+    ("origStr" in fb && "newStr" in fb) ||
+    (respData["placeSource"] == "DEFAULT" &&
+      respData["pastSourceContext"] != "Earth")
+  );
+}
+
+//
+// For more context, refer:
+//   https://docs.google.com/document/d/13-1zsYZm_RFkzhKb4GxVy7DQNU8ZiEYZSY8001R-GsU/edit
+//
+export function NLCommentary(props: NLCommentaryPropType): JSX.Element {
+  return (
+    <>
+      {!props.chartsData.placeFallback &&
```

Comment:
strongly recommend to have a few child component like:

```
export function A1(props: chartsData): JSX.Element {
  showA := <complicated && ||>
  return <>(showA? <div/>)</>
}
A2...
A3...
```

Then can have meaningful name for A1, A2..

Reply:
Thanks for the offline suggestion to factor the "control" flow separate from the display strings. PTAL now.